readalongs/align.py

```diff
@@ -420,11 +420,11 @@ def frames_to_time(frames):
     }
     silence_offsets = defaultdict(int)
     silence = 0
-    if results["tokenized"].xpath("//*[@silence]"):
+    if results["tokenized"].xpath("//silence"):
         endpoint = 0
         for el in results["tokenized"].xpath("//*"):
-            if "silence" in el.attrib:
-                silence_ms = parse_time(el.attrib["silence"])
+            if el.tag == "silence" and "dur" in el.attrib:
+                silence_ms = parse_time(el.attrib["dur"])
```

Comment:
This needs a try/except KeyError to gracefully handle bad inputs. With anchors, I made it a hard error with what I tried to make a clear error message (see `get_sequences()`).

I'm torn between making the silence one a warning or a similarly hard error. Warning would issue a message but ignore the silence request. That's seemingly friendlier, in the spirit of "do something reasonable with the input given", but it would also lead to frustrating troubleshooting, à la "I put a silence in there, why does my read-along not have that silence?", which can be prevented with a hard error instead.

I vote hard error.